cli.js

```diff
@@ -35,29 +32,64 @@ var cli = meow({
 	string: ['_']
 });
 
+var errors = [];
+var failed = 0;
+var passed = 0;
+
 function error(err) {
 	console.error(err.stack);
 	process.exit(1);
 }
 
+function test(data) {
+	var isError = data.err.message;
+
+	if (isError) {
+		log.error(data.title, chalk.red(data.err.message));
+
+		errors.push(data);
+		failed++;
+	} else {
+		log.test(null, data.title, data.duration);
+
+		passed++;
+	}
+}
+
 function run(file) {
-	fs.stat(file, function (err, stats) {
-		if (err) {
-			console.error(err.message);
-			process.exit(1);
-		}
-
-		if (stats.isDirectory()) {
-			init(path.join(file, '*.js'));
-			return;
-		}
-
-		if (path.extname(file) !== '.js') {
-			return;
-		}
-
-		require(file);
+	var stats = fs.statSync(file);
+
+	if (stats.isDirectory()) {
+		return init(path.join(file, '*.js'));
+	}
+
+	if (path.extname(file) !== '.js') {
+		return Promise.resolve();
+	}
+
+	return fork(file).on('message', test);
```

Comment:
Don't understand. This function returns a promise. It also emits `message` event on completed/failed test, like we agreed earlier. Returned promise resolves, when all tests are completed.